src/shared/parse-helpers.c

```diff
@@ -196,3 +198,33 @@ int parse_socket_bind_item(
         *port_min = mn;
         return 0;
 }
+
+int parse_open_file_fields(const char *v, char **path, char **fdname, int *flags) {
+        _cleanup_strv_free_ char **parts = NULL;
+        char *tmp_path = NULL;
+        char *tmp_fdname = NULL;
+        int r;
+
+        if (!v)
+                return -EINVAL;
+
+        r = strv_split_full(&parts, v, ":", EXTRACT_DONT_COALESCE_SEPARATORS);
+        if (r < 0)
+                return r;
+        if (strv_length(parts) != 3)
```

Comment:
Ditto that we shouldn't require all three parts to be specified all the time, see my earlier comment